### PR TITLE
Add new Pi-hole diagnostics system

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Add a new tool providing users with easily accessible error messages. Previously, all these massages were only accessible by looking carefully through `/var/log/pihole-FTL.log`. I assume a lot of issues stay unnoticed. 

Whenever FTL wants to send a message to the user, a new (clickable) icon appears
Screenshot at 2020-05-11 20-38-36

The small label already indicates how many issues have been found in the local configuration. A click on this icon leads to a new page providing further details

Screenshot_2020-05-11 Pi-hole - ubuntu-server(1)

**How does this PR accomplish the above?:**

Check and display messages from the `message` table in `pihole-FTL.db`.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: scripts/pi-hole/js/messages.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/* Pi-hole: A black hole for Internet advertisements
+ *  (c) 2017 Pi-hole, LLC (https://pi-hole.net)
+ *  Network-wide ad blocking via your own hardware.
+ *
+ *  This file is copyright under the latest version of the EUPL.
+ *  Please see LICENSE file for your rights under this license. */
+
+/* global utils:false */
+
+var token = $("#token").html();
```
Comment (XhmikosR) [inline scripts/pi-hole/js/messages.js]: `text()`

File: style/pi-hole.css [hunk 2]
```diff
@@ -310,4 +310,13 @@ code.breakall
 .bootstrap-select.bs-container.align-right {
     left: unset !important;
     right: 10px;
+}
+
+.warning {
+    color: orange;
```
Comment (XhmikosR) [inline style/pi-hole.css]: I'd use the `text-orange` class to keep CSS simple.

File: style/pi-hole.css [hunk 3]
```diff
@@ -310,4 +310,13 @@ code.breakall
...
+    font-size: 18px;
+}
+
+.hidden {
```
Comment (XhmikosR) [inline style/pi-hole.css]: I'd rename this to `d-none` and put a reminder to re-use the class in the whole codebase instead of inline CSS or JS code.

**Comment by DL6ER** [comment]
Two notes when you want to try this locally before it is merged:

1. You'll also need https://github.com/pi-hole/FTL/pull/752
2. You'll only see REGEX warnings as of now. Even when this web PR already supports SUBNET warnings as well, we first have to wait for https://github.com/pi-hole/FTL/pull/753 being merged before we can add the necessary backend code for this.

I tested it carefully and it works for me.

File: style/pi-hole.css (hunk 3 above)
Comment (XhmikosR) [reply style/pi-hole.css]: @DL6ER you missed adding the needed CSS for `d-none` :)

**Comment by yubiuser** [comment]
For me "Settings/API_Web Interface" is broken with this branch
Bildschirmfoto zu 2020-05-12 12-30-14

**Comment by dschaper** [comment]
`<<<<HEAD` means there's a merge conflict that wasn't resolved. Did you do any `git` commands?

**Comment by yubiuser** [comment]
[Once](https://discourse.pi-hole.net/t/dark-mode-for-web-interface/29878/21?u=yubiuser ) after
```
[i] Update repo in /var/www/html/admin... 
: Could not update local repository. Contact support.
```
I did 

```
cd /var/www/html/admin
sudo git reset --hard HEAD^^
sudo git pull
```
everything is working since then.

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/subnetting-is-not-working-with-most-relevant-entry/31482/16

**Comment by yubiuser** [comment]
Can confirm the extraneous characters are gone.

**Comment by DL6ER** [comment]
Resolved merge conflicts.

**Comment by PromoFaux** [comment]
Can confirm the page loads, but I have no errors :D

@DL6ER, I'm not feeling very inventive, can you suggest some way that I can force an error to prove the functionality pleases?

**Comment by DL6ER** [comment]
@promofaux Add invalid regex such as `[^` and `(((`

**Comment by PromoFaux** [comment]
OK, yeah, that's v nice

**Review by PromoFaux — APPROVED** [review]

**Comment by XhmikosR** [comment]
@DL6ER Just confirming, was Network supposed to be moved in the Tools menu item?

**Comment by DL6ER** [comment]
@XhmikosR Yes, this page is less relevant for the main purpose of Pi-hole (DNS service) and only there for debugging purposes. Hence, I moved it into the section where the other diagnostics tools are as well.

**Pull request merged**